### ISSUE TITLE
Error: foreach() argument must be of type array|object, null given on click to confirm secondary email

### ISSUE THREAD

**Issue by AndyXheli** [nextcloud/server#29022]
### Steps to reproduce
1. Add a second email
2. Click on verify link
### Server configuration

**Operating system:**
Ubuntu 20.4
**Web server:**
Apache 2
**Database:**
Maria DB 10.5.12
**PHP version:**
8
**Nextcloud version:** (see Nextcloud admin page)
22.2.0
**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from Nc 22.1.0

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:**  Yes

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"aq1qLk9zdm5GXckpJ6Ye","level":3,"time":"2021-10-01T16:17:30+00:00","remoteAddr":"66.158.36.34","user":"admin","app":"PHP","method":"GET","url":"/apps/provisioning_api/mailVerification/4f4b141842fffb8ceadc7f3a0583cb970030e9004a2b4128c076d47749702def%7C22402c9efe13bc74a944e2db62b96d74%7C69e34c531d048cf5696b006fe45f7d6769964c5ae57d59bffe6f7febd265a68c1d551fccf003613ff5640ee6f642b7aaca0a29eb21de5540e2418452d288a2e5%7C3/J2pzeG7z4XiecbdqjVfns/admin","message":"foreach() argument must be of type array|object, null given at /var/www/nextcloud/core/templates/confirmation.php#15","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.61 Safari/537.36","version":"22.2.0.2","exception":{"Exception":"Error","Message":"foreach() argument must be of type array|object, null given at /var/www/nextcloud/core/templates/confirmation.php#15","Code":0,"Trace":[{"file":"/var/www/nextcloud/core/templates/confirmation.php","line":15,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":180,"args":["/var/www/nextcloud/core/templates/confirmation.php"],"function":"include"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":150,"function":"load","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Template.php","line":179,"function":"fetchPage","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/public/AppFramework/Http/TemplateResponse.php","line":204,"function":"fetchPage","class":"OC_Template","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":171,"function":"render","class":"OCP\\AppFramework\\Http\\TemplateResponse","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":156,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":301,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1000,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"id":"6157349aa328d"}
{"reqId":"aq1qLk9zdm5GXckpJ6Ye","level":3,"time":"2021-10-01T16:17:30+00:00","remoteAddr":"66.158.36.34","user":"admin","app":"PHP","method":"GET","url":"/apps/provisioning_api/mailVerification/4f4b141842fffb8ceadc7f3a0583cb970030e9004a2b4128c076d47749702def%7C22402c9efe13bc74a944e2db62b96d74%7C69e34c531d048cf5696b006fe45f7d6769964c5ae57d59bffe6f7febd265a68c1d551fccf003613ff5640ee6f642b7aaca0a29eb21de5540e2418452d288a2e5%7C3/J2pzeG7z4XiecbdqjVfns/admin","message":"Undefined array key \"parameters\" at /var/www/nextcloud/core/templates/confirmation.php#15","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.61 Safari/537.36","version":"22.2.0.2","exception":{"Exception":"Error","Message":"Undefined array key \"parameters\" at /var/www/nextcloud/core/templates/confirmation.php#15","Code":0,"Trace":[{"file":"/var/www/nextcloud/core/templates/confirmation.php","line":15,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":180,"args":["/var/www/nextcloud/core/templates/confirmation.php"],"function":"include"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":150,"function":"load","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Template.php","line":179,"function":"fetchPage","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/public/AppFramework/Http/TemplateResponse.php","line":204,"function":"fetchPage","class":"OC_Template","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":171,"function":"render","class":"OCP\\AppFramework\\Http\\TemplateResponse","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":156,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":301,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1000,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"id":"6157349aa32cb"}

{"reqId":"aq1qLk9zdm5GXckpJ6Ye","level":3,"time":"2021-10-01T16:17:30+00:00","remoteAddr":"66.158.36.34","user":"admin","app":"PHP","method":"GET","url":"/apps/provisioning_api/mailVerification/4f4b141842fffb8ceadc7f3a0583cb970030e9004a2b4128c076d47749702def%7C22402c9efe13bc74a944e2db62b96d74%7C69e34c531d048cf5696b006fe45f7d6769964c5ae57d59bffe6f7febd265a68c1d551fccf003613ff5640ee6f642b7aaca0a29eb21de5540e2418452d288a2e5%7C3/J2pzeG7z4XiecbdqjVfns/admin","message":"Undefined array key \"targetUrl\" at /var/www/nextcloud/core/templates/confirmation.php#9","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.61 Safari/537.36","version":"22.2.0.2","exception":{"Exception":"Error","Message":"Undefined array key \"targetUrl\" at /var/www/nextcloud/core/templates/confirmation.php#9","Code":0,"Trace":[{"file":"/var/www/nextcloud/core/templates/confirmation.php","line":9,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":180,"args":["/var/www/nextcloud/core/templates/confirmation.php"],"function":"include"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":150,"function":"load","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Template.php","line":179,"function":"fetchPage","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/public/AppFramework/Http/TemplateResponse.php","line":204,"function":"fetchPage","class":"OC_Template","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":171,"function":"render","class":"OCP\\AppFramework\\Http\\TemplateResponse","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":156,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":301,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1000,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"id":"6157349aa3305"}


```
</details>

**Comment by andyxheli** [comment]
Still an issue on NC 25.0.3

**Comment by joshtrichards** [comment]
Fixed in #37303
